tracer.go

```diff
@@ -73,6 +73,12 @@ type Options struct {
 	// When set, it attempts to exacerbate issues emanating from use of Spans
 	// after calling Finish by running additional assertions.
 	DebugAssertUseAfterFinish bool
+	// DisableSpanPool disables reuse of the underlying span structs.
+	// Buyer Beware, any operations called on a span after Finish is undefined
+	// behavior. The main usecase of disabling the pool is to allow the creation
```

Comment:
I agree with @jmacd that `EnableSpanPool`, defaulted to false, is preferable. It's not really going to break people's code because the default mode will be _safer_, but slightly less performant.